plugin/documents.py

```diff
@@ -252,10 +252,13 @@ def diagnostics_intersecting_region_async(
     ) -> Tuple[List[Tuple[SessionBuffer, List[Diagnostic]]], sublime.Region]:
         covering = sublime.Region(region.a, region.b)
         result = []  # type: List[Tuple[SessionBuffer, List[Diagnostic]]]
+        requested_range = region_to_range(self.view, region)
         for sb, diagnostics in self.diagnostics_async():
             intersections = []  # type: List[Diagnostic]
             for diagnostic, candidate in diagnostics:
-                if region.intersects(candidate):
+                # Perform intersection checks on Range since it's inclusive unlike Region which is exclusive.
+                diagnostic_range = region_to_range(self.view, candidate)
+                if requested_range.intersects(diagnostic_range):
                     covering = covering.cover(candidate)
                     intersections.append(diagnostic)
             if intersections:
```

Comment:
This whole function (and related) could probably be refactored to work on `Range` exclusively but that's too much of a job for now.